components/pluralrules/src/lib.rs

```diff
@@ -0,0 +1,381 @@
+    /// [`data provider`]: ./data/trait.DataProviderType.html
+    pub fn try_new<D: data::provider::DataProviderType>(
+        locale: LanguageIdentifier,
+        type_: PluralRuleType,
+        data_provider: &D,
+    ) -> Result<Self, PluralRulesError> {
+        let selector = data_provider
+            .get_selector(&locale, type_)?
+            .ok_or(PluralRulesError::MissingData)?;
+        Ok(Self { locale, selector })
+    }
+
+    /// Returns the [`Plural Category`] appropriate for the given number.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_locale::LanguageIdentifier;
+    /// use icu_pluralrules::{PluralRules, PluralRuleType, PluralCategory};
+    /// use icu_pluralrules::data::provider::DummyDataProvider;
+    ///
+    /// let lang: LanguageIdentifier = "en".parse()
+    ///     .expect("Failed to parse a language identifier.");
+    ///
+    /// let dp = DummyDataProvider::default();
+    ///
+    /// let pr = PluralRules::try_new(lang, PluralRuleType::Cardinal, &dp)
+    ///     .expect("Failed to construct a PluralRules struct.");
+    ///
+    /// match pr.select(1_usize) {
+    ///     PluralCategory::One => "One item",
+    ///     PluralCategory::Other => "Many items",
+    ///     _ => { unreachable!(); }
+    /// };
+    /// ```
+    ///
+    /// The method accepts any input that can be calculated into [`Plural Operands`].
+    /// All unsigned primitive number types can infallibly be converted so they can be
+    /// used as an input.
+    ///
+    /// For signed numbers and strings, [`Plural Operands`] implement `TryFrom` and `FromStr`,
+    /// which should be used before passing the result to `select`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use std::convert::TryFrom;
+    /// # use icu_locale::LanguageIdentifier;
+    /// # use icu_pluralrules::{PluralRules, PluralRuleType};
+    /// use icu_pluralrules::{PluralCategory, PluralOperands};
+    /// # use icu_pluralrules::data::provider::DummyDataProvider;
+    /// #
+    /// # let lang: LanguageIdentifier = "en".parse()
+    /// #     .expect("Failed to parse a language identifier.");
+    /// #
+    /// # let dp = DummyDataProvider::default();
+    /// #
+    /// # let pr = PluralRules::try_new(lang, PluralRuleType::Cardinal, &dp)
+    /// #     .expect("Failed to construct a PluralRules struct.");
+    ///
+    /// let operands = PluralOperands::try_from(-5.2)
+    ///     .expect("Failed to parse to operands.");
+    /// let operands2: PluralOperands = "5.10".parse()
+    ///     .expect("Failed to parse to operands.");
+    ///
+    /// assert_eq!(pr.select(operands), PluralCategory::Other);
+    /// assert_eq!(pr.select(operands2), PluralCategory::Other);
+    /// ```
+    ///
+    /// [`Plural Category`]: ./enum.PluralCategory.html
+    /// [`Plural Operands`]: ./operands/struct.PluralOperands.html
+    pub fn select<I: Into<PluralOperands>>(&self, input: I) -> PluralCategory {
```

Comment:
Why `Into<PluralOperands>` instead of `&PluralOperands`?  I'm not suggesting you change it; I'm just wondering what led to that design decision.

Reply:
Because that results in a lower ergonomic of conversions in simple cases like `pr.select(5_usize)`

With the current signature, anything that can be infallibly converted into operands will work, but if I turn it into anything that can be references as operands, I can't convert implicitly because I need to allocate somewhere.

Reply:
FixedDecimal doesn't implement Copy, but it will be able to `impl From<&FixedDecimal> for PluralOperands`.

Will you be able to select based on a FixedDecimal without cloning or moving the FixedDecimal?

Reply:
If you only have `impl From<&FixedDecimal> for PluralOperands` then the user will be able to do `pr.select(&fd)`.
If you also add ``impl From<FixedDecimal> for PluralOperands` then the user will also be able to do `pr.select(fd)`